### PR TITLE
[Health] Expand Micro Nutrient Support

### PR DESCRIPTION
**Pull request by davidpryor**
Hello! I am not sure about the contribution guidelines as they seem to have been deleted or moved locations: https://github.com/cph-cachet/flutter-plugins/CONTRIBUTING.md

## Description
This PR aims to expand the Health package to support micro nutrients for reading and writing from Health Connect, Google Fit (to be deprecated), and Apple Health.

Nutrients we added based on the health platforms' documentation
https://developer.apple.com/documentation/healthkit/data_types/nutrition_type_identifiers
https://developer.android.com/reference/kotlin/androidx/health/connect/client/records/NutritionRecord
https://developers.google.com/fit/scenarios/add-nutrition-data

My intentions we to avoid any API breaking changes. I did do some clean-up on the method channel API as there seemed to be a mix of snake_case and camelCase being used.

I did change the `NutritionHealthValue.fromHealthDataPoint` factory, which may not be desirable, but I a open to changing that if desired.

It also seems like the ios plugin was only reading a single food in the getData method, so I also updated that code to grab all foods and return them.

### CODE REVIEW

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 1]
```diff
@@ -1153,16 +1195,48 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
-                forIdentifier: .dietaryEnergyConsumed)!
-            dataTypesDict[DIETARY_CAFFEINE] = HKSampleType.quantityType(
-                forIdentifier: .dietaryCaffeine)!
-            dataTypesDict[DIETARY_FATS_CONSUMED] = HKSampleType.quantityType(
-                forIdentifier: .dietaryFatTotal)!
-            dataTypesDict[DIETARY_PROTEIN_CONSUMED] = HKSampleType.quantityType(
-                forIdentifier: .dietaryProtein)!
+            
+            // Nutrition
+            dataTypesDict[DIETARY_CARBS_CONSUMED] = HKSampleType.quantityType(forIdentifier: .dietaryCarbohydrates)!
```
Comment (davidpryor) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift:1200]: All of the micronutrients are available in ios 8+ besides dietaryWater which is ios 9+
I believe the plugin only differentiates for ios 11 and 13, so it should be safe to add these always

**Comment by Panosfunk** [comment]
Hello @davidpryor, 
since Google Fit is to be deprecated we want to start removing its support from the plugin.
Therefore this needs to be removed not only from this PR before it is merged, but from the entire plugin. It would save us a lot of time if this is something you could look into.

**Comment by davidpryor** [comment]
> Hello @davidpryor, since Google Fit is to be deprecated we want to start removing its support from the plugin. Therefore this needs to be removed not only from this PR before it is merged, but from the entire plugin. It would save us a lot of time if this is something you could look into.

I went ahead and rolled back the changes to writeMeal for the Google Fit implementation.

**Comment by davidpryor** [comment]
> Hey @davidpryor, if you're willing to do the same for the entirety of the health package that would be of great help, if not then we are still very grateful for your contribution to the package!

I can look into stripping it out, but would prefer it be a separate PR.

**Comment by Panosfunk** [comment]
Yes, that would be better.

**Comment by davidpryor** [comment]
@Panosfunk sounds good! I believe I removed google fit related code from this PR. I will look into removing google fit in a new PR soon.

**Comment by SlimShadyIAm** [comment]
Hi @davidpryor, have you already begun working on removing Google Fit? If not I can start to look into it.

**Review by SlimShadyIAm — APPROVED** [review]

**Comment by davidpryor** [comment]
> Hi @davidpryor, have you already begun working on removing Google Fit? If not I can start to look into it.

Sorry for the delayed response. I have not and won't actually be able to focus on it any time soon, unfortunately

**Comment by SlimShadyIAm** [comment]
No worries! Thanks for your contribution :)

**Pull request merged**